core/src/main/scala/tech/beshu/ror/accesscontrol/factory/BlockValidator.scala

```diff
@@ -74,51 +78,77 @@ object BlockValidator {
+                                                               kibanaRulesInBlock: NonEmptyList[KibanaRelatedRule],
+                                                               allRulesInBlock: NonEmptyList[Rule]) = {
+    NonEmptyList.fromList {
+      allRulesInBlock.toList.flatMap(validateRuleUsageInContextOf(blockName, kibanaRulesInBlock))
+    } match {
+      case Some(errors) => Validated.Invalid(errors)
+      case None => Validated.Valid(())
+    }
   }
 
-  private def validateIfKibanaRelatedRulesCoexistenceWithOther(rules: NonEmptyList[RuleDefinition[Rule]]) = {
-    if(isKibanaRelated(rules)) {
-      NonEmptyList
-        .fromList {
-          rules
-            .map(_.rule)
-            .collect[BlockValidationError] {
-              case _: ActionsRule => KibanaRuleTogetherWith.ActionsRule
-              case _: FilterRule => KibanaRuleTogetherWith.FilterRule
-              case _: FieldsRule => KibanaRuleTogetherWith.FieldsRule
-              case _: ResponseFieldsRule => KibanaRuleTogetherWith.ResponseFieldsRule
-            }
-        } match {
-        case None => Validated.Valid(())
-        case Some(errors) => Validated.Invalid(errors)
+  private def validateRuleUsageInContextOf(blockName: Block.Name,
+                                           kibanaRules: NonEmptyList[KibanaRelatedRule]): Rule => Option[KibanaRuleTogetherWith] = {
+    case _: ActionsRule =>
+      determineConfiguredKibanaAccessIn(blockName, kibanaRules) match {
+        case None | Some(KibanaAccess.Unrestricted) => None
+        case Some(_) => Some(KibanaRuleTogetherWith.ActionsRule)
+      }
+    case _: FilterRule =>
+      Some(KibanaRuleTogetherWith.FilterRule)
+    case _: FieldsRule =>
+      Some(KibanaRuleTogetherWith.FieldsRule)
+    case _: ResponseFieldsRule =>
+      Some(KibanaRuleTogetherWith.ResponseFieldsRule)
+    case _ =>
+      None
+  }
+
+  private def findKibanaRelatedRules(rules: NonEmptyList[Rule]): Option[NonEmptyList[KibanaRelatedRule]] = {
+    NonEmptyList.fromList {
+      rules.collect {
+        case r: KibanaRelatedRule => r
       }
-    } else {
-      Validated.Valid(())
     }
   }
 
-  private def validateIfKibanaUserDataRuleIsNotUsedWithOldDeprecatedKibanaRules(ruleDefs: NonEmptyList[RuleDefinition[Rule]]) = {
-    if(containsKibanaUserDataRule(ruleDefs)) {
+  private def determineConfiguredKibanaAccessIn(blockName: Block.Name,
+                                                kibanaRules: NonEmptyList[KibanaRelatedRule]) = {
+    kibanaRules.collect {
+      case r: KibanaAccessRule => r.settings.access
+      case r: KibanaUserDataRule => r.settings.access
+    } match {
+      case Nil =>
+        None
+      case head :: Nil =>
+        Some(head)
+      case head :: _ =>
+        throw new IllegalStateException(s"More than one kibana access rule found in the '${blockName.show}'! It may lead to unexpected behaviors. Please, report this problem as soon as possible.'")
```

Comment:
I'm probably missing something obvious, but it is not clear to me where the issue should be reported to